featurebyte/service/base_document.py

```diff
@@ -84,11 +85,15 @@ class BaseDocumentService(
 
     document_class: Type[Document]
 
-    def __init__(self, user: Any, persistent: Persistent, catalog_id: ObjectId):
+    def __init__(self, user: Any, persistent: Persistent, catalog_id: Optional[ObjectId]):
         self.user = user
         self.persistent = persistent
         self.catalog_id = catalog_id
         self._allow_to_use_raw_query_filter = False
+        if self.is_catalog_specific and not catalog_id:
+            raise CatalogNotSpecifiedError(
```

Comment:
Is this error case covered in the tests?

Reply:
I think not. Will add a test

Reply:
Added tests for both service and route